### ISSUE TITLE
GLM-V4.1thing 在8卡4090,每张48g显存上比4卡2080ti，每张22g推理更慢

### ISSUE THREAD

**Issue by dyhuachi** [zai-org/GLM-V#202]
### System Info / 系統信息

4090配置：

<img width="832" height="960" alt="Image" src="https://github.com/user-attachments/assets/9ff1d35d-a518-47a0-bf0f-afe818d030b9" />
(base) root@iZ0jlff0fvxeix66ormjcsZ:~/GLM-4.1V-Thinking/inference# cat /proc/cpuinfo| grep "physical id"| sort| uniq| wc -l
2
(base) root@iZ0jlff0fvxeix66ormjcsZ:~/GLM-4.1V-Thinking/inference# cat /proc/cpuinfo| grep "cpu cores"| uniq
cpu cores	: 32

2080Ti配置：

<img width="853" height="630" alt="Image" src="https://github.com/user-attachments/assets/0655b822-f528-402f-912b-df8ab4fa242d" />
(base) dieu@ubuntu-server:~$ cat /proc/cpuinfo| grep "physical id"| sort| uniq| wc -l
2
(base) dieu@ubuntu-server:~$ cat /proc/cpuinfo| grep "cpu cores"| uniq
cpu cores	: 1

代码：
# app.py
import io
import torch
from PIL import Image
from fastapi import FastAPI, File, UploadFile, HTTPException
from fastapi.responses import JSONResponse
from transformers import AutoProcessor, Glm4vForConditionalGeneration
import re
import time

# ================== 全局模型加载（服务启动时加载一次）==================
MODEL_PATH = "/run/user/0/models/ZhipuAI/GLM-4.1V-9B-Thinking"

print("[INFO] Loading model and processor...")
processor = AutoProcessor.from_pretrained(MODEL_PATH, use_fast=True)
model = Glm4vForConditionalGeneration.from_pretrained(
    MODEL_PATH,
    torch_dtype=torch.float16,
    device_map="auto"
)
print("[INFO] Model loaded successfully.")

app = FastAPI(title="GLM-4V Animal State Classifier", version="1.0")

PROMPT = (
'"输出一个状态标签，从以下五类中选择:出笼治疗：不存在动物则直接输出这个标签;休息：动物存在且躺着坐着趴着状态直接输出这个标签;进食：动物头部接触盆里的水或者粮食直接输出这个标签;输液：动物身上有透明输液管直接输出这个标签;活动：动物存在且不属于前面的标签直接输出这个标签;如果你反复思考则直接输出为活动标签"'
)

@app.post("/predict")
async def predict(file: UploadFile = File(...)):
    if not file.content_type.startswith("image/"):
        raise HTTPException(status_code=400, detail="File must be an image.")
    
    start_time = time.time()
    
    try:
        # 读取图像
        image_bytes = await file.read()
        image = Image.open(io.BytesIO(image_bytes)).convert("RGB")
        
        # 构建消息
        messages = [{
            "role": "user",
            "content": [
                {"type": "image", "image": image},  # 注意：这里传 PIL.Image 对象
                {"type": "text", "text": PROMPT}
            ]
        }]
        
        # 预处理
        inputs = processor.apply_chat_template(
            messages,
            tokenize=True,
            add_generation_prompt=True,
            return_dict=True,
            return_tensors="pt",
            padding=True,
        ).to(model.device)
        
        # 推理
        with torch.inference_mode():
            output = model.generate(
                **inputs,
                max_new_tokens=512,
                do_sample=False,
            )
        
        # 解码
        raw_output = processor.decode(
            output[0][inputs["input_ids"].shape[1]:],
            skip_special_tokens=False
        )
        
        # 提取 answer
        match = re.search(r"<answer>(.*?)</answer>", raw_output, re.DOTALL)
        answer = match.group(1).strip() if match else raw_output.strip()
        
        total_time = time.time() - start_time
        
        return JSONResponse({
            "status": "success",
            "answer": answer,
            "raw_output": raw_output,
            "processing_time_sec": round(total_time, 2),
            "model_device": str(model.device) if hasattr(model, 'device') else "multi-gpu"
        })
    
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Prediction error: {str(e)}")

@app.get("/health")
async def health_check():
    return {"status": "ok", "model_loaded": True}

调用代码：
# client.py
import requests

url = "http://localhost:8000/predict"
image_path = "./03.png"

with open(image_path, "rb") as f:
    files = {"file": (image_path, f, "image/png")}
    response = requests.post(url, files=files)

if response.status_code == 200:
    result = response.json()
    print("✅ Answer:", result["answer"])
    print("⏱️  Time:", result["processing_time_sec"], "seconds")
    print("🖥️  Device:", result["model_device"])
else:
    print("❌ Error:", response.json())

图片：

![Image](https://github.com/user-attachments/assets/9323745b-8fde-49ac-8354-534602ad6cf1)
<img width="1111" height="648" alt="Image" src="https://github.com/user-attachments/assets/9d4dc9d0-3d07-4c18-a74c-f3a442d325cf" />

### Information / 问题信息

- [ ] The official example scripts / 官方的示例脚本
- [ ] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

按照官网下载requirements.txt包 然后run这个服务启动代码和测试代码；
4090上彩色图片90s，那张夜视图片要30s
2080ti上彩色图片8s，夜视图片19s
官网api调用彩图在3s，夜视图片6s

### Expected behavior / 期待表现

期待您给出答案解答我的困惑

**Comment by zRzRzRzRzRzRzR** [comment]
你用transformers载入是把一个模型分到了8张卡和4张卡, 而不是并行, 所以更慢
你可以用vLLM 或者sglang这样的框架试试,这些推引擎自己也有速度测试工具

**Comment by dyhuachi** [comment]
> 你用transformers载入是把一个模型分到了8张卡和4张卡, 而不是并行, 所以更慢 你可以用vLLM 或者sglang这样的框架试试,这些推引擎自己也有速度测试工具

ok，但是我在使用vllm的时候遇到了困惑，希望得到帮助
